graph.py

```diff
@@ -46,23 +91,14 @@ def update_links(self, links):
             if link.is_active():
                 self.graph.add_edge(link.endpoint_a.id, link.endpoint_b.id)
                 for key, value in link.metadata.items():
-                    keys.extend(key)
+                    keys.append(key)
```

Comment:
This is indeed important for me to explain. I will get back to you on this.

Reply:
Hello @hdiogenes, I looked at the bodies of the HTTP POST responses after searching for the best paths between the same two points using the current algorithm, which uses extend, and this PR's algorithm, which uses append. They are the same:

Current algorithm results:
![Screenshot from 2020-07-03 14-52-30](https://user-images.githubusercontent.com/26232289/86493210-998d0780-bd3e-11ea-96d2-523fc8cc9055.png)

PR algorithm results:
![Screenshot from 2020-07-03 14-55-14](https://user-images.githubusercontent.com/26232289/86493217-a0b41580-bd3e-11ea-9124-64bf8d681f66.png)

This feels like good news but it does not answer your question, since append and extend work differently and the change might affect something I missed.

I will keep looking into this.

Reply:
I have reverted this change.

Reply:
I talked with David. He said that append was causing each character in the key string to be added to keys instead of the whole string. Changing it to `extend` fixed this.